core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -649,26 +649,27 @@ private[group] class GroupMetadata(val groupId: String, initialState: GroupState
     }
   }
 
-  def failPendingOffsetWrite(topicPartition: TopicPartition, offset: OffsetAndMetadata): Unit = {
+  def failPendingOffsetWrite(topicIdPartition: TopicIdPartition, offset: OffsetAndMetadata): Unit = {
+    val topicPartition = topicIdPartition.topicPartition
     pendingOffsetCommits.get(topicPartition) match {
       case Some(pendingOffset) if offset == pendingOffset => pendingOffsetCommits.remove(topicPartition)
       case _ =>
     }
   }
 
-  def prepareOffsetCommit(offsets: Map[TopicPartition, OffsetAndMetadata]): Unit = {
+  def prepareOffsetCommit(offsets: Map[TopicIdPartition, OffsetAndMetadata]): Unit = {
     receivedConsumerOffsetCommits = true
-    pendingOffsetCommits ++= offsets
+    pendingOffsetCommits ++= offsets.map { case(k, v) => k.topicPartition -> v }
```

Comment:
nit: I wonder what is best here: using ++= (addAll) or iterating over offsets and use +=. The latter does not create an intermediate collection. What do you think?
nit2: `case (k, v)`. We put a space after `case`.